### PR TITLE
fast query chunk by coord

### PR DESCRIPTION
**Pull request by zCri**

### CODE REVIEW

File: nylium.Core/World/Chunk.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public class Chunk {
         public B.Block[][][] Blocks { get; }
 
         public Chunk(World parent, int x, int z) {
+            parent.Chunks.TryAdd((x, z), this);
```
Comment (dskprt) [inline nylium.Core/World/Chunk.cs:19]: Remove this as the chunk generation in World.cs also adds it to the list

File: nylium.Core/World/Chunk.cs [hunk 2]
```diff
@@ -25,6 +26,7 @@ public class Chunk {
         }
 
         public Chunk(World parent, int x, int z, B.Block[][][] blocks) {
+            parent.Chunks.TryAdd((x, z), this);
```
Comment (dskprt) [inline nylium.Core/World/Chunk.cs:29]: Remove this too

File: nylium.Core/World/World.cs [hunk 3]
```diff
@@ -77,18 +79,11 @@ public class World {
 
         // TODO better way to do this?
         public Chunk GetChunk(int x, int z) {
-            for(int i = 0; i < Chunks.Count; i++) {
-                Chunk chunk = Chunks[i];
-
-                if(chunk.X == x && chunk.Z == z) {
-                    return chunk;
-                }
+            try {
```
Comment (dskprt) [inline nylium.Core/World/World.cs:82]: It would be better to check whether the entry exists or not with .ContainsKey()

Comment (zCri) [reply nylium.Core/World/World.cs:82]: why

Comment (zCri) [reply nylium.Core/World/World.cs:82]: this is fine

**Pull request closed without merging**